pkg/controller/controller_binding.go

```diff
@@ -95,12 +104,27 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 			binding.Namespace,
 			binding.Name,
 		)
-		return nil
+		return true, nil
 	}
 
-	toUpdate, err := makeServiceInstanceCredentialClone(binding)
-	if err != nil {
-		return err
+	// If the reconciler has exceeded the time allowed to retry failed
+	// reconciliations on the ServiceInstanceCredential, then do not reconcile it.
+	if binding.Status.OperationStartTime != nil && !time.Now().Before(binding.Status.OperationStartTime.Time.Add(c.reconciliationRetryDuration)) {
```

Comment:
Important:

I think the logic here needs to be a little different.  The test on this line should determine whether true or false is returned if there's an error, ie, whether the current reconciliation is the 'final chance'.

When the final chance fails, we should ensure that the ready condition is false and that it has a message that tells the user the operation failed.  I think that the failed condition should be set for instances and credentials.  We don't yet have that for brokers, but this might be time to add it.

Reply:
Makes sense. I made the change.